server/src/main/java/org/opensearch/action/admin/indices/segments/TransportPitSegmentsAction.java

```diff
@@ -0,0 +1,260 @@
+            transportService,
+            actionFilters,
+            indexNameExpressionResolver,
+            PitSegmentsRequest::new,
+            ThreadPool.Names.MANAGEMENT
+        );
+        this.clusterService = clusterService;
+        this.indicesService = indicesService;
+        this.searchService = searchService;
+        this.namedWriteableRegistry = namedWriteableRegistry;
+        this.transportService = transportService;
+        this.pitService = pitService;
+    }
+
+    /**
+     * Execute PIT segments flow for all PITs or request PIT IDs
+     */
+    @Override
+    protected void doExecute(Task task, PitSegmentsRequest request, ActionListener<IndicesSegmentResponse> listener) {
+        if (request.getPitIds().isEmpty()) {
+            pitService.getAllPits(ActionListener.wrap(response -> {
+                request.setPitIds(response.getPitInfos().stream().map(ListPitInfo::getPitId).collect(Collectors.toList()));
+                super.doExecute(task, request, listener);
+            }, listener::onFailure));
+        } else {
+            super.doExecute(task, request, listener);
+        }
+    }
+
+    /**
+     * This adds list of shards on which we need to retrieve pit segments details
+     * @param clusterState    the cluster state
+     * @param request         the underlying request
+     * @param concreteIndices the concrete indices on which to execute the operation
+     */
+    @Override
+    protected ShardsIterator shards(ClusterState clusterState, PitSegmentsRequest request, String[] concreteIndices) {
+        final ArrayList<ShardRouting> iterators = new ArrayList<>();
+        for (String pitId : request.getPitIds()) {
+            SearchContextId searchContext = decode(namedWriteableRegistry, pitId);
+            for (Map.Entry<ShardId, SearchContextIdForNode> entry : searchContext.shards().entrySet()) {
+                final SearchContextIdForNode perNode = entry.getValue();
+                // check if node is part of local cluster
+                if (Strings.isEmpty(perNode.getClusterAlias())) {
+                    final ShardId shardId = entry.getKey();
+                    iterators.add(
+                        new PitAwareShardRouting(
+                            pitId,
+                            shardId,
+                            perNode.getNode(),
+                            null,
+                            true,
+                            ShardRoutingState.STARTED,
+                            null,
+                            null,
+                            null,
+                            -1L
+                        )
+                    );
+                }
+            }
+        }
+        return new PlainShardsIterator(iterators);
+    }
+
+    @Override
+    protected ClusterBlockException checkGlobalBlock(ClusterState state, PitSegmentsRequest request) {
+        return state.blocks().globalBlockedException(ClusterBlockLevel.METADATA_READ);
+    }
```

Comment:
should this be `READ` instead of `METADATA_READ`?

Reply:
`METADATA_READ` is used in index segments as well. `Read` is used in operations like search, multisearch etc.